### PR TITLE
feat (csv): support for csv files

### PR DESCRIPTION
**Pull request by enudler**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -26,15 +26,15 @@ jobs:
           path: ./coverage
   integration-tests:
     docker:
-      - image: circleci/node:10
+      - image: circleci/node:12
```
Comment (NivLipetz) [inline .circleci/config.yml:29]: Please also update `Dockerfile` node version to node:12

File: tests/test-scripts/id_to_name.csv [hunk 2]
```diff
@@ -0,0 +1,5 @@
+id,name
+1,eli
+2,mikcey
```
Comment (NivLipetz) [inline tests/test-scripts/id_to_name.csv:3]: <3

File: tests/unit-tests/connectors/fileDownloadConnector-test.js [hunk 3]
```diff
@@ -3,9 +3,7 @@ const should = require('should'),
     uuid = require('uuid');
 
 const request = require('../../../app/helpers/requestSender'),
-    customJSConnector = require('../../../app/connectors/customJSConnector'),
-    testUtils = require('../../utils/consts');
-
+    customJSConnector = require('../../../app/connectors/fileDownloadConnector');
```
Comment (NivLipetz) [inline tests/unit-tests/connectors/fileDownloadConnector-test.js:6]:
```suggestion
    fileConector = require('../../../app/connectors/fileDownloadConnector');
```

File: tests/unit-tests/helpers/file_processor_positive-test.js [hunk 4]
```diff
@@ -1,9 +1,5 @@
-const rewire = require('rewire'),
-    should = require('should'),
-    fs = require('fs'),
-    runner = rewire('../../../app/models/runner'),
-    path = require('path'),
-    fileConector = require('../../../app/connectors/customJSConnector'),
+const should = require('should'),
+    fileConector = require('../../../app/connectors/fileDownloadConnector'),
```
Comment (NivLipetz) [inline tests/unit-tests/helpers/file_processor_positive-test.js:2]:
```suggestion
    fileConnector = require('../../../app/connectors/fileDownloadConnector'),
```

File: app/models/runner.js [hunk 5]
```diff
@@ -1,8 +1,9 @@
 'use strict';
 
 const artillery = require('artillery/core'),
+    csv = require('csv-parse/lib/sync'),
     testFileConnector = require('../connectors/testFileConnector'),
-    customJSConnector = require('../connectors/customJSConnector'),
+    fileDownloadConnector = require('../connectors/fileDownloadConnector'),
```
Comment (NivLipetz) [inline app/models/runner.js:6]:
```suggestion
    fileConnector = require('../connectors/fileDownloadConnector'),
```

File: patch.json [hunk 6]
```diff
@@ -0,0 +1,738 @@
+Index: app/connectors/customJSConnector.js
```
Comment (NivLipetz) [inline patch.json:1]: remove file

File: processor_file.js [hunk 7]
```diff
@@ -0,0 +1,3 @@
+let addFunction = (a, b) => {
```
Comment (NivLipetz) [inline processor_file.js:1]: remove file

File: tests/test-scripts/id_to_name.csv (hunk 2 above)
Comment (NivLipetz) [inline tests/test-scripts/id_to_name.csv:3]:
```suggestion
2,mickey
```

File: tests/unit-tests/models/runTest-test.js [hunk 8]
```diff
@@ -233,6 +233,62 @@ describe('Run test', () => {
...
+        let testWithCSV = Object.assign({}, consts.VALID_CUSTOM_TEST);
+        testWithCSV.csv_file_id = csvFileId;
+
+        testFileConnectorStub.resolves(testWithCSV);
+        artilleryStub.resolves(ee);
+        reporterConnectorCreateReportStub.resolves();
+        reporterConnectorPostStatsStub.resolves();
+        getFileStub.resolves('id,name\n' +
+            '1,eli\n' +
+            '2,mikcey\n' +
```
Comment (NivLipetz) [inline tests/unit-tests/models/runTest-test.js:255]:
```suggestion
            '2,mickey\n' +
```

File: tests/unit-tests/models/runTest-test.js [hunk 9]
```diff
@@ -233,6 +233,62 @@ describe('Run test', () => {
...
+            'name'
+        ]);
+        artilleryArgs[1].should.eql([
+            [
+                '1',
+                'eli'
+            ],
+            [
+                '2',
+                'mikcey'
```
Comment (NivLipetz) [inline tests/unit-tests/models/runTest-test.js:279]:
```suggestion
                'mickey'
```

File: .circleci/config.yml (hunk 1 above)
Comment (enudler) [reply .circleci/config.yml:29]: put it back to 10
will update another time.
don't want to run to issues now

**Pull request merged**